engine/source_sym_engine.ml

```diff
@@ -170,15 +178,20 @@ let group_constructors type_env (acs, rows) : (constructor * matrix) list * matr
         end;
         collect_constructors ptl
   in
-  List.iter (fun row -> collect_constructors row.lhs) rows;
+  List.iter (fun row -> collect_constructors (fst row.lhs)) rows;
   let all_constructor_groups =
     group_tbl |> Hashtbl.to_seq_values |> List.of_seq
   in
   let rec put_in_group (row : matrix_row) =
-    match row.lhs with
+    let patterns, env = row.lhs in
+    let env = match patterns with
+      | (As (_, var))::_ -> SMap.add var (SAccessor (List.hd acs)) env
+      | _ -> env
+    in
+    match patterns with
```

Comment:
This code is not quite right because you look for a single `As` at the head, and then you traverse the head pattern structure below, ignoring the `As` construct. For example `(_ as x) as y` or `(K1 as x | K2 as y)` would not get handled correctly with this logic. Instead you should add an `env` parameter to the `put_in_group` function, and extend it in the `As` case with a recursive call below.